library/Icingadb/Widget/JoystickPaginator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Icinga\Module\Icingadb\Widget;
+
+use ipl\Orm\Query;
+
+class JoystickPaginator
```

Comment:
I guess you added this as an optimization? The thing is, I don't quite see a justification for this being outsourced. The joystick paginator is subject to change with the new style for the servicegrid, so this is also likely subject to change as well. I think, leaving the calculations in the view script is fine for now. :wink: